light_sdk/src/tx_builders/pos.rs

```diff
@@ -0,0 +1,653 @@
+// Redelegate transaction WASM path
+const TX_REDELEGATE_WASM: &str = "tx_redelegate.wasm";
+// Change validator metadata WASM path
+const TX_CHANGE_METADATA_WASM: &str = "tx_change_validator_metadata.wasm";
+// Change consensus key WASM path
+const TX_CHANGE_CONSENSUS_KEY_WASM: &str = "tx_change_consensus_key.wasm";
+// Change commission WASM path
+const TX_CHANGE_COMMISSION_WASM: &str = "tx_change_validator_commission.wasm";
+// Withdraw WASM path
+const TX_WITHDRAW_WASM: &str = "tx_withdraw.wasm";
+
+/// A bond transaction
+pub struct Bond(Tx);
+
+impl Bond {
+    /// Build a raw Bond transaction from the given parameters
+    pub fn new(
+        validator: Address,
+        amount: token::Amount,
+        source: Option<Address>,
+        timestamp: DateTimeUtc,
+        expiration: Option<DateTimeUtc>,
+        code_hash: Hash,
+        chain_id: ChainId,
+    ) -> Self {
+        let unbond = namada_core::types::transaction::pos::Bond {
+            validator,
+            amount,
+            source,
+        };
+
+        Self(tx_builders::build_tx(
+            unbond,
+            timestamp,
+            expiration,
+            code_hash,
+            TX_BOND_WASM.to_string(),
+            chain_id,
+        ))
+    }
+
+    //FIXME: share
+    /// Generate the signature(s) for the given transaction and signers
+    pub fn generate_signatures(
+        mut self,
+        secret_keys: &[common::SecretKey],
+        public_keys_index_map: &AccountPublicKeysMap,
+        signer: Option<Address>,
+    ) -> (Self, Vec<SignatureIndex>) {
+        let (tx, sigs) = tx_builders::generate_tx_signatures(
+            self.0,
+            secret_keys,
+            public_keys_index_map,
+            signer,
+        );
+
+        (Self(tx), sigs)
+    }
+
+    //FIXME: share
+    /// Attach the provided signatures to the tx
+    pub fn attach_signatures(
+        mut self,
+        signatures: Vec<SignatureIndex>,
+    ) -> Self {
+        Self(tx_builders::attach_raw_signatures(self.0, signatures))
+    }
+}
+
+/// An unbond transaction
+pub struct Unbond(Tx);
+
+impl Unbond {
+    /// Build a raw Unbond transaction from the given parameters
+    pub fn new(
+        validator: Address,
+        amount: token::Amount,
+        source: Option<Address>,
+        timestamp: DateTimeUtc,
+        expiration: Option<DateTimeUtc>,
+        code_hash: Hash,
+        chain_id: ChainId,
+    ) -> Self {
+        let unbond = namada_core::types::transaction::pos::Unbond {
+            validator,
+            amount,
+            source,
+        };
+
+        Self(tx_builders::build_tx(
+            unbond,
+            timestamp,
+            expiration,
+            code_hash,
+            TX_UNBOND_WASM.to_string(),
+            chain_id,
+        ))
+    }
+
+    /// Generate the signature(s) for the given transaction and signers
+    pub fn generate_signatures(
+        mut self,
+        secret_keys: &[common::SecretKey],
```

Comment:
the library doesn't take secret keys. 

it should expose `sign_bytes()` and then the caller figures out how to sign it